### ISSUE TITLE
Update checker should support downgrade to more stable version if I change update Channel

### ISSUE THREAD

**Issue by unv-unv** [consulo/consulo#311]
- IDE build number: 1905
- Java version: 1.8
- OS name and version: Win10

If I change update channel then update checker should download most recent update for each plugin in this channel regardless if plugin version is less than installed one.

For example, I experience problems with plugins installed via Nigtly channel (Java plugin wasn't updated and I cannot work with my code). I go to Settings > General > Platform and Plugins > Web Services > Channel, change it to "release" and expect that update checker would download latest stable build for all the plugins. This doesn't happen (since update checker plainly compares plugin versions and my installed nightly plugins are definitely more recent than stable ones) and IDE stays unusable.

**Comment by VISTALL** [comment]
Yeah, it's good idea too. With plugin updates - it's easy to fix. But with platform harder. Consulo launcher iterate **platform** directory and select higher directory (natural sort).

Looks like need store some information in text file (it's need rewrite sh scripts, C++ win launcher, ObjC mac launcher) - ouch


Thanks

**Comment by unv-unv** [comment]
One more time nightly builds became unusable. Many exceptions from Java plugin in Event Log, on trying to submit them HTTP 500 error is reported. Problems with Java imports, syntax highlighting, etc.

So I needed to rollback to stable builds. And I've spent about a hour doing so because I couldn't downgrade by simply selecting stable build channel and checking for update and had a funny time deleting files in various directories until it started to work again.

So please consider implementing such a downgrade when switching between channels. If it's a problem to select correct platform by channel, maybe simple deletion of irrelevant platform versions from disk may help? After all, it's not a problem to download someting one more time.

**Comment by VISTALL** [comment]
Good day. Yeah - nightly became unstable, due i'm working at migration to jre11 (big refactoring too).

Downgrade will be nice, but for that need change boot logic. Current implementation select higher build from platform directory, that why - if you place *lower* build it will ignored. Need implement that in C++ code(windows), ObjC(mac), sh (unix)

> maybe simple deletion of irrelevant platform versions from disk may help?

There few moments. Better replace our distribution by release distribution, and need drop last build from platform directory(https://github.com/consulo/consulo/wiki/Paths-to-system-directories).
That need - since, Consulo extract build builds to custom directory (that need - since distribution may stored in admin area, and run not as admin). And run check for updates (will be restore plugin builds)

Since i am use nightly builds too - i always fix almost all issues. If you have some issues - will be nice, if you notify about that(https://discuss.consulo.io/).

**Comment by unv-unv** [comment]
As you know, deadline always comes unpredictably, so it was no time to wait for fixes :)
